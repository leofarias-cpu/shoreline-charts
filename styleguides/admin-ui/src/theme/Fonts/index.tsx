import React, { memo } from 'react'
import { Global, css } from '@emotion/core'

// import VariableWoff from './variable.woff'
import VariableWoff2 from './variable.woff2'

/**
 * Defines & applies VTEX Trust font
 */
function Fonts() {
  return (
    <Global
      styles={css`
        @font-face {
          font-family: 'VTEX Trust';
          src: local('VTEX Trust'), local('VTEX Trust'),
            url(${VariableWoff2}) format('woff2-variations'),
          font-weight: 1 999;
          font-style: normal;
        }
        html, * {
          font-family: 'VTEX Trust', sans-serif;
        }
      `}
    />
  )
}

export default memo(Fonts)
