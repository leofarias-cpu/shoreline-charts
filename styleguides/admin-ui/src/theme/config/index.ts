import layout from './layout'
import colors from './colors'
import space from './space'
import sizes from './sizes'
// components
import buttons from './buttons'
import text from './text'
import toggle from './toggle'
import avatar from './avatar'
import overlay from './overlay'
import checkbox from './checkbox'
import divider from './divider'
import controlGroup from './controlGroup'
import radio from './radio'
import collapsible from './collapsible'
import data from './data'

export const theme = {
  ...layout,
  space,
  sizes,
  colors,
  buttons,
  avatar,
  overlay,
  data,
  divider,
  collapsible,
  data,
  forms: {
    ...controlGroup,
    ...toggle,
    ...checkbox,
    ...radio,
  },
  text,
}

export type FontSizes = '0' | '1' | '2' | '3' | '4'
export type BorderWidths = '0' | '1' | '2' | '3'
export type BorderRadius = '0' | '1' | '2' | '3' | 'full'
export type FontWeights = keyof typeof theme.fontWeights
export type LineHeights = keyof typeof theme.lineHeights
export type ZIndexes = 'under' | 'plain' | 'over' | '1' | '2' | '3' | '4' | '5'

export type Space =
  | '0'
  | '1'
  | '2'
  | '3'
  | '4'
  | '5'
  | '6'
  | '7'
  | '8'
  | 'px'
  | '2px'

export type Sizes = keyof typeof sizes

export type AdminTheme = typeof theme
