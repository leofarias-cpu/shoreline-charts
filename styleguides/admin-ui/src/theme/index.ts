export { ThemeProvider } from './Provider'
export { useTheme, ThemeContext } from './useTheme'
export {
  theme,
  Theme,
  Colors,
  FeedbackPalette,
  FeedbackPalettes,
} from './theme'
