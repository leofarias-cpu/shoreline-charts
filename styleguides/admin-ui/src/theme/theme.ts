export const theme = {
  space: {
    '0': '0rem',
    '1': '0.0625rem',
    '2': '0.125rem',
    '3': '0.25rem',
    '4': '0.375rem',
    '5': '0.5rem',
    '6': '0.625rem',
    '7': '0.75rem',
    '8': '0.875rem',
    '9': '1rem',
    '10': '1.125rem',
    '11': '1.25rem',
    '12': '1.375rem',
    '13': '1.5rem',
  },
  sizes: {
    '0': '0rem',
    '1': '0.0625rem',
    '2': '0.125rem',
    '3': '0.25rem',
    '4': '0.5rem',
    '5': '0.75rem',
    '6': '1rem',
    '7': '1.25rem',
    '8': '1.5rem',
    '9': '1.75rem',
    '10': '2rem',
    '11': '2.25rem',
    '12': '2.5rem',
    '13': '3rem',
    '14': '4rem',
    '15': '5rem',
    '16': '6rem',
    '17': '8rem',
    '18': '10rem',
    '19': '12rem',
    '20': '14rem',
    '21': '16rem',
    xs: '20rem',
    sm: '24rem',
    md: '28rem',
    lg: '32rem',
    xl: '36rem',
    '2xl': '42rem',
    '3xl': '48rem',
    '4xl': '56rem',
    '5xl': '64rem',
    '6xl': '72rem',
    '1/2': '50%',
    '1/3': '33.333333%',
    '2/3': '66.666667%',
    '1/4': '25%',
    '2/4': '50%',
    '3/4': '75%',
    '1/5': '20%',
    '2/5': '40%',
    '3/5': '60%',
    '4/5': '80%',
    '1/6': '16.666667%',
    '2/6': '33.333333%',
    '3/6': '50%',
    '4/6': '66.666667%',
    '5/6': '83.333333%',
    '1/12': '8.333333%',
    '2/12': '16.666667%',
    '3/12': '25%',
    '4/12': '33.333333%',
    '5/12': '41.666667%',
    '6/12': '50%',
    '7/12': '58.333333%',
    '8/12': '66.666667%',
    '9/12': '75%',
    '10/12': '83.333333%',
    '11/12': '91.666667%',
    full: '100%',
    screenHeight: '100vh',
    screenWidth: '100vw',
  },
  colors: {
    text: '#323845',
    background: '#FFFFFF',
    muted: ['#686E7B', '#8B9299', '#C1C6CC', '#D7DADF', '#F3F5F9'],
    focus: '#8DB6FA',
    primary: {
      base: '#2953B2',
      hover: '#1E4397',
      active: '#3F6FDB',
      contrast: '#FFFFFF',
      washed: '#F4F8FE',
    },
    danger: {
      base: '#D23030',
      hover: '#A70C0C',
      active: '#DE0404',
      contrast: '#FFFFFF',
      washed: '#FFF0F0',
    },
  },
  components: {
    skeleton: {
      styles: {
        display: 'inline-block',
        width: 'full',
        height: 'full',
        backgroundColor: 'muted.4',
        backgroundSize: `200px 100%`,
        backgroundRepeat: 'no-repeat',
        lineHeight: 1,
      },
      shape: {
        rect: {
          borderRadius: 4,
        },
        circle: {
          borderRadius: '100%',
        },
      },
    },
    button: {
      styles: {
        borderWidth: 1,
        borderRadius: 3,
        cursor: 'pointer',
        position: 'relative',
        borderStyle: 'solid',
        '&:focus': {
          outline: 'none',
        },
      },
      size: {
        regular: {
          paddingY: 5,
          fontSize: 1,
          height: 12,
          width: 'auto',
          paddingLeft: 9,
          paddingRight: 9,
        },
        'regular-icon': {
          paddingY: 5,
          fontSize: 1,
          height: 12,
          width: 12,
          paddingLeft: 3,
          paddingRight: 3,
        },
        'regular-icon-start': {
          paddingY: 5,
          fontSize: 1,
          height: 12,
          width: 'auto',
          paddingLeft: 5,
          paddingRight: 7,
        },
        'regular-icon-end': {
          paddingY: 5,
          fontSize: 1,
          height: 12,
          width: 'auto',
          paddingLeft: 7,
          paddingRight: 5,
        },
        small: {
          paddingY: 4,
          fontSize: 0,
          height: 10,
          width: 'auto',
          paddingLeft: 7,
          paddingRight: 7,
        },
        'small-icon': {
          paddingY: 4,
          fontSize: 0,
          height: 10,
          width: 10,
          paddingLeft: 2,
          paddingRight: 2,
        },
        'small-icon-start': {
          paddingY: 4,
          fontSize: 0,
          height: 10,
          width: 'auto',
          paddingLeft: 5,
          paddingRight: 7,
        },
        'small-icon-end': {
          paddingY: 4,
          fontSize: 0,
          height: 10,
          width: 'auto',
          paddingLeft: 7,
          paddingRight: 5,
        },
      },
      variant: {
        'filled-primary': {
          textTransform: 'uppercase',
          color: 'primary.contrast',
          backgroundColor: 'primary.base',
          fontVariationSettings: "'wght' 92",
          borderColor: 'primary.base',
          '&:hover': {
            backgroundColor: 'primary.hover',
            borderColor: 'primary.hover',
          },
          '&:active': {
            backgroundColor: 'primary.active',
            borderColor: 'primary.active',
          },
          '&:disabled': {
            color: 'text',
            borderColor: 'muted.2',
            backgroundColor: 'muted.2',
          },
        },
        'outlined-primary': {
          textTransform: 'uppercase',
          backgroundColor: 'transparent',
          borderColor: 'muted.2',
          color: 'primary.base',
          fontVariationSettings: "'wght' 92",
          '&:hover': {
            backgroundColor: 'primary.washed',
            color: 'primary.hover',
          },
          '&:active': {
            color: 'primary.active',
          },
          '&:disabled': {
            color: 'muted.1',
            backgroundColor: 'muted.4',
          },
        },
        'subtle-primary': {
          borderColor: 'transparent',
          textTransform: 'capitalize',
          backgroundColor: 'transparent',
          color: 'primary.base',
          fontVariationSettings: "'wght' 92",
          '&:hover': {
            color: 'primary.hover',
            backgroundColor: 'primary.washed',
          },
          '&:active': {
            color: 'primary.active',
            backgroundColor: 'primary.washed',
          },
          '&:disabled': {
            color: 'muted.1',
          },
        },
        'filled-danger': {
          textTransform: 'uppercase',
          color: 'danger.contrast',
          backgroundColor: 'danger.base',
          fontVariationSettings: "'wght' 92",
          borderColor: 'danger.base',
          '&:hover': {
            borderColor: 'danger.hover',
            backgroundColor: 'danger.hover',
          },
          '&:active': {
            borderColor: 'danger.active',
            backgroundColor: 'danger.active',
          },
          '&:disabled': {
            borderColor: 'muted.2',
            color: 'text',
            backgroundColor: 'muted.2',
          },
        },
        'outlined-danger': {
          textTransform: 'uppercase',
          backgroundColor: 'transparent',
          borderColor: 'muted.2',
          color: 'danger.base',
          fontVariationSettings: "'wght' 92",
          '&:hover': {
            backgroundColor: 'danger.washed',
            color: 'danger.hover',
          },
          '&:active': {
            color: 'danger.active',
          },
          '&:disabled': {
            color: 'muted.1',
            backgroundColor: 'muted.4',
          },
        },
        'subtle-danger': {
          borderColor: 'transparent',
          textTransform: 'capitalize',
          backgroundColor: 'transparent',
          color: 'danger.base',
          fontVariationSettings: "'wght' 92",
          '&:hover': {
            color: 'danger.hover',
            backgroundColor: 'danger.washed',
          },
          '&:active': {
            color: 'danger.active',
            backgroundColor: 'danger.washed',
          },
          '&:disabled': {
            color: 'muted.1',
          },
        },
      },
    },
    text: {
      variant: {
        small: {
          lineHeight: 'small',
          fontVariationSettings: "'wght' 80",
          fontSize: 0,
        },
        body: {
          lineHeight: 'body',
          fontVariationSettings: "'wght' 80",
          fontSize: 1,
        },
        highlight: {
          lineHeight: 'highlight',
          fontVariationSettings: "'wght' 92",
          fontSize: 1,
        },
        action: {
          lineHeight: 'action',
          fontVariationSettings: "'wght' 92",
          fontSize: 1,
          textTransform: 'uppercase',
        },
        subtitle: {
          lineHeight: 'subtitle',
          fontVariationSettings: "'wght' 92",
          fontSize: 2,
        },
        headline: {
          lineHeight: 'headline',
          fontVariationSettings: "'wght' 92",
          fontSize: 3,
        },
      },
    },
  },
  fontSizes: [12, 14, 16, 20],
  lineHeights: {
    small: 1.125,
    body: 1.25,
    highlight: 1.25,
    action: 1.5,
    subtitle: 1.5,
    headline: 1.5,
  },
  borderWidths: [0, 1, 2, 4, 6],
  borderRadius: [0, 1, 2, 4, 6],
}

export type AdminTheme = typeof theme
