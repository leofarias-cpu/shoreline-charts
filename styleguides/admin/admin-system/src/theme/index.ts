export { contrast } from './util'
export { default as ThemeProvider } from './Provider'
export { darken, lighten, desaturate, saturate, rgba } from 'polished'
export * from './config'
export * from './hooks'
