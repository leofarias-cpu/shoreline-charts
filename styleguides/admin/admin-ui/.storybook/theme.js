import { create } from '@storybook/theming/create'
import { theme } from '@vtex/admin-ui-theme'

const { colors, borderRadius } = theme

export default create({
  base: 'light',

  colorPrimary: colors.primary.base,
  colorSecondary: colors.text,

  // UI
  appBg: colors.background,
  appContentBg: colors.muted[4],
  appBorderColor: colors.muted[3],
  appBorderRadius: borderRadius[3],

  // Typography
  fontBase:
    '"VTEX Trust Variable", -apple-system, system-ui, BlinkMacSystemFont, sans-serif',
  fontCode:
    '"Dank Mono", "Operator Mono", "Fira Code Retina", "Fira Code", "FiraCode-Retina", "Consolas", "Monaco", monospace',

  // Text colors
  textColor: colors.text,
  textInverseColor: colors.background,

  // Toolbar default and active colors
  barTextColor: colors.background,
  barSelectedColor: colors.background,
  barBg: colors.text,

  // Form colors
  inputBg: colors.background,
  inputBorder: colors.muted[3],
  inputTextColor: colors.text,
  inputBorderRadius: borderRadius[3],

  brandTitle: 'Admin UI',
  brandUrl: 'https://github.com/vtex/onda',
  brandImage: 'http://brand.vtex.com/static/media/logo.2f3fc60b.svg',
})
