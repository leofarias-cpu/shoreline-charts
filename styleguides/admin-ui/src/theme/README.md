# Admin UI Theme

## ThemeProvider

> Provider of all admin-ui styles

The ThemeProvider component is a wrapper around Emotion's ThemeProvider and MDX's MDXProvider. It provides default styled-components to MDX content that pick up values from `theme.styles`. It also handles color mode state.

Usage:

```jsx
import { ThemeProvider } from '@vtex/admin-ui'

function ProjectRoot() {
  return <ThemeProvider>{/** You app here */}</ThemeProvider>
}
```

## jsx

The jsx export is a React create element function intended for use with a custom pragma comment. It adds support for the sx prop, which uses Emotion's create element function internally and parses style objects with the Theme UI css utility.

Usage:

```jsx
/** @jsx jsx */
import { jsx } from '@vtex/admin-ui'

export default (props) => (
  <div
    {...props}
    sx={{
      color: 'primary.base',
    }}
  />
)
```

## `useTheme` hook

> Returns the theme object

Usage:

```jsx
import { useTheme } from '@vtex-components/theme'

function Component() {
  const theme = useTheme()
  // ...
}
```

## `sx` property

> To apply custom styles and consume theme values on specyfic properties

The sx prop lets you add any valid CSS to an element, while using values from your theme to keep styles consistent. You can think of the style object that the sx prop accepts as a superset of CSS.

Some property are theme-aware, that can consume specyfic sections of the theme context. You can see the full token list down below.

[Read more on ThemeUI Documentation](https://theme-ui.com/sx-prop)

## Token List

> List of all tokens and props that can consume it

### Space

Supported props:

```
margin, margin-top, margin-right, margin-bottom, margin-left, margin-block, margin-block-end, margin-block-start, margin-inline, margin-inline-end, margin-inline-start, padding, padding-top, padding-right, padding-bottom, padding-left, padding-block, padding-block-end, padding-block-start, padding-inline, padding-inline-end, padding-inline-start, inset, inset-block, inset-block-end, inset-block-start, inset-inline, inset-inline-end, inset-inline-start, top, right, bottom, left, grid-gap, grid-column-gap, grid-row-gap, gap, column-gap, row-gap
```

| token | value  |
| ----- | ------ |
| 0     | `0px`  |
| 1     | `1px`  |
| 2     | `2px`  |
| 3     | `4px`  |
| 4     | `6px`  |
| 5     | `8px`  |
| 6     | `10px` |
| 7     | `12px` |
| 8     | `14px` |
| 9     | `16px` |
| 10    | `18px` |
| 11    | `20px` |
| 12    | `22px` |
| 13    | `24px` |

### Color

Supported props:

```
color, background-color, border-color, caret-color, border-top-color, border-bottom-color, border-left-color, border-right-color, outline-color, fill, stroke
```

| token              | value     |
| ------------------ | --------- |
| `text`             | '#323845' |
| `background`       | '#FFFFFF' |
| `focus`            | '#8DB6FA' |
| `muted.0`          | '#686E7B' |
| `muted.1`          | '#8B9299' |
| `muted.2`          | '#C1C6CC' |
| `muted.3`          | '#D7DADF' |
| `muted.4`          | '#F3F5F9' |
| `primary.base`     | '#2953B2' |
| `primary.hover`    | '#1E4397' |
| `primary.active`   | '#3F6FDB' |
| `primary.contrast` | '#FFFFFF' |
| `primary.washed`   | '#F4F8FE' |
| `danger.base`      | '#D23030' |
| `danger.hover`     | '#A70C0C' |
| `danger.active`    | '#DE0404' |
| `danger.contrast`  | '#FFFFFF' |
| `danger.washed`    | '#FFF0F0' |

### fontSize

Supported props:

```
font-size
```

| token | value  |
| ----- | ------ |
| 0     | `12px` |
| 1     | `14px` |
| 2     | `16px` |
| 3     | `20px` |

### Line Height

Supported props:

```
line-height
```

| token       | value     |
| ----------- | --------- |
| `small`     | `1.125em` |
| `body`      | `1.125em` |
| `highlight` | `1.125em` |
| `action`    | `1.5em`   |
| `subtitle`  | `1.5em`   |
| `headline`  | `1.5em`   |

### Border Width

Supported props:

```
border-width, border-top-width, border-bottom-width, border-left-width, border-right-width, border-block-end-width, border-block-start-width, border-block-width, border-inline-end-width, border-inline-start-width, border-inline-width
```

| token | value |
| ----- | ----- |
| 0     | `0px` |
| 1     | `1px` |
| 2     | `2px` |
| 3     | `4px` |
| 4     | `6px` |

### Border Radius

Supported props:

```
border-radius, border-top-right-radius, border-top-left-radius, border-bottom-right-radius, border-bottom-left-radius, border-end-end-radius, border-end-start-radius, border-start-end-radius, border-start-start-radius
```

| token | value |
| ----- | ----- |
| 0     | `0px` |
| 1     | `1px` |
| 2     | `2px` |
| 3     | `4px` |
| 4     | `6px` |
