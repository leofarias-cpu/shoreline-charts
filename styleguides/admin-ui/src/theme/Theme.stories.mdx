import { Meta, Story, Canvas } from '@storybook/addon-docs/blocks'

<Meta title="design-system/Theme" />

# Admin UI Theme

> ðŸ’¡ Also check [ThemeUI API](https://theme-ui.com/api/) for detailed info.

## ThemeProvider

> Same as [ThemeUI ThemeProvider](https://theme-ui.com/guides/how-it-works/#themeprovider)

Usage:

```jsx
import { ThemeProvider } from '@vtex/admin-ui'

function ProjectRoot() {
  return <ThemeProvider>{/** Your app here */}</ThemeProvider>
}
```

## jsx

> Same as [JSX Pragma](https://theme-ui.com/guides/how-it-works/#jsx-pragma)

Usage:

```jsx
/** @jsx jsx */
import { jsx } from '@vtex/admin-ui'

export default (props) => (
  <div
    {...props}
    sx={{
      color: 'primary.base',
    }}
  />
)
```

## `useTheme` hook

> Returns the theme object

Usage:

```jsx
import { useTheme } from '@vtex-components/theme'

function Component() {
  const theme = useTheme()
  // ...
}
```

## `sx` property

> To apply custom styles and consume theme values on specyfic properties

The sx prop lets you add any valid CSS to an element, while using values from your theme to keep styles consistent. You can think of the style object that the sx prop accepts as a superset of CSS.

Some property are theme-aware, that can consume specyfic sections of the theme context. You can see the full token list down below.

[Read more on ThemeUI Documentation](https://theme-ui.com/sx-prop)

## Styled Props

### Color

```js
function useCase() {
  return <Box bg="primary.base" c="primary.contrast" />
}
```

#### Available Tokens

| Token    | CSS Property          | Theme Field |
| -------- | --------------------- | ----------- |
| `c`      | `color`               | `colors`    |
| `bg`     | `background-color`    | `colors`    |
| `cc`     | `caret-color`         | `colors`    |
| `bc`     | `border-color`        | `colors`    |
| `btc`    | `border-top-color`    | `colors`    |
| `brc`    | `border-right-color`  | `colors`    |
| `bbc`    | `border-bottom-color` | `colors`    |
| `blc`    | `border-left-color`   | `colors`    |
| `oc`     | `outline-color`       | `colors`    |
| `fill`   | `fill`                | `colors`    |
| `stroke` | `stroke`              | `colors`    |

#### Values

| Token Value        | Returned Value | Color Preview                                                                                                       |
| ------------------ | -------------- | ------------------------------------------------------------------------------------------------------------------- |
| `text`             | #323845        | <span style={{ borderRadius: '4px', display: 'flex', backgroundColor:'#323845', height: '20px', width: '100%' }} /> |
| `background`       | #FFFFFF        | <span style={{ borderRadius: '4px', display: 'flex', backgroundColor:'#FFFFFF', height: '20px', width: '100%' }} /> |
| `focus`            | #8DB6FA        | <span style={{ borderRadius: '4px', display: 'flex', backgroundColor:'#8DB6FA', height: '20px', width: '100%' }} /> |
| `emphasis`         | #F71963        | <span style={{ borderRadius: '4px', display: 'flex', backgroundColor:'#F71963', height: '20px', width: '100%' }} /> |
| `muted.0`          | #686E7B        | <span style={{ borderRadius: '4px', display: 'flex', backgroundColor:'#686E7B', height: '20px', width: '100%' }} /> |
| `muted.1`          | #8B9299        | <span style={{ borderRadius: '4px', display: 'flex', backgroundColor:'#8B9299', height: '20px', width: '100%' }} /> |
| `muted.2`          | #C1C6CC        | <span style={{ borderRadius: '4px', display: 'flex', backgroundColor:'#C1C6CC', height: '20px', width: '100%' }} /> |
| `muted.3`          | #D7DADF        | <span style={{ borderRadius: '4px', display: 'flex', backgroundColor:'#D7DADF', height: '20px', width: '100%' }} /> |
| `muted.4`          | #F3F5F9        | <span style={{ borderRadius: '4px', display: 'flex', backgroundColor:'#F3F5F9', height: '20px', width: '100%' }} /> |
| `primary.base`     | #2953B2        | <span style={{ borderRadius: '4px', display: 'flex', backgroundColor:'#2953B2', height: '20px', width: '100%' }} /> |
| `primary.hover`    | #1E4397        | <span style={{ borderRadius: '4px', display: 'flex', backgroundColor:'#1E4397', height: '20px', width: '100%' }} /> |
| `primary.active`   | #3F6FDB        | <span style={{ borderRadius: '4px', display: 'flex', backgroundColor:'#3F6FDB', height: '20px', width: '100%' }} /> |
| `primary.contrast` | #FFFFFF        | <span style={{ borderRadius: '4px', display: 'flex', backgroundColor:'#FFFFFF', height: '20px', width: '100%' }} /> |
| `primary.washed`   | #F4F8FE        | <span style={{ borderRadius: '4px', display: 'flex', backgroundColor:'#F4F8FE', height: '20px', width: '100%' }} /> |
| `danger.base`      | #D23030        | <span style={{ borderRadius: '4px', display: 'flex', backgroundColor:'#D23030', height: '20px', width: '100%' }} /> |
| `danger.hover`     | #A70C0C        | <span style={{ borderRadius: '4px', display: 'flex', backgroundColor:'#A70C0C', height: '20px', width: '100%' }} /> |
| `danger.active`    | #DE0404        | <span style={{ borderRadius: '4px', display: 'flex', backgroundColor:'#DE0404', height: '20px', width: '100%' }} /> |
| `danger.contrast`  | #FFFFFF        | <span style={{ borderRadius: '4px', display: 'flex', backgroundColor:'#FFFFFF', height: '20px', width: '100%' }} /> |
| `danger.washed`    | #FFF0F0        | <span style={{ borderRadius: '4px', display: 'flex', backgroundColor:'#FFF0F0', height: '20px', width: '100%' }} /> |

### Space

```js
function useCase() {
  return <Box m="2" pb="4" />
}
```

#### Available Tokens

| Token | CSS Property                       | Theme Field |
| ----- | ---------------------------------- | ----------- |
| `m`   | `margin`                           | `space`     |
| `mt`  | `margin-top`                       | `space`     |
| `mr`  | `margin-right`                     | `space`     |
| `mb`  | `margin-bottom`                    | `space`     |
| `ml`  | `margin-left`                      | `space`     |
| `mx`  | `margin-left` and `margin-right`   | `space`     |
| `my`  | `margin-top` and `margin-bottom`   | `space`     |
| `p`   | `padding`                          | `space`     |
| `pt`  | `padding-top`                      | `space`     |
| `pr`  | `padding-right`                    | `space`     |
| `pb`  | `padding-bottom`                   | `space`     |
| `pl`  | `padding-left`                     | `space`     |
| `px`  | `padding-left` and `padding-right` | `space`     |
| `py`  | `padding-top` and `padding-bottom` | `space`     |

#### Values

| Token Value | Returned Value |
| ----------- | -------------- |
| 0           | `0rem`         |
| 1           | `0.0625rem`    |
| 2           | `0.125rem`     |
| 3           | `0.25rem`      |
| 4           | `0.375rem`     |
| 5           | `0.5rem`       |
| 6           | `0.625rem`     |
| 7           | `0.75rem`      |
| 8           | `0.875rem`     |
| 9           | `1rem`         |
| 10          | `1.125rem`     |
| 11          | `1.25rem`      |
| 12          | `1.375rem`     |
| 13          | `1.5rem`       |
| 14          | `1.625rem`     |
| 15          | `1.875rem`     |
| 16          | `2rem`         |
| 17          | `2.125rem`     |
| 18          | `2.25rem`      |
| 19          | `2.375rem`     |
| 20          | `2.5rem`       |
| 21          | `2.625rem`     |
| 22          | `2.75rem`      |
| 23          | `2.875rem`     |
| 24          | `3rem`         |

### Flex

```js
function useCase() {
  return <Box width="full" height="2xl" overflow="hidden" />
}
```

#### Available Tokens

| Prop        | CSS Property      | Theme Field | Values                                       |
| ----------- | ----------------- | ----------- | -------------------------------------------- |
| `items`     | `align-items`     | none        | start, end, center, baseline, stretch, auto  |
| `justify`   | `justify-content` | none        | start, end, center, between, around, evenly  |
| `wrap`      | `flex-wrap`       | none        | wrap, reverse, nowrap                        |
| `direction` | `flex-direction`  | none        | row, rowReverse, col, colReverse             |
| `self`      | `align-self`      | none        | start, end, center, baseline, stretch, auto  |
| `content`   | `align-content`   | none        | start, end, center, between, around, stretch |
| `grow`      | `flex-grow`       | none        | number                                       |
| `shrink`    | `flex-shrink`     | none        | number                                       |
| `order`     | `order`           | none        | number                                       |

### Grid

```jsx
function useCase() {
  return <Box gridGap="2" gridAutoFlow="row dense" />
}
```

| Prop                  | CSS Property            | Theme Field |
| --------------------- | ----------------------- | ----------- |
| `gridGap`             | `grid-gap`              | `space`     |
| `gridRowGap`          | `grid-row-gap`          | `space`     |
| `gridColumnGap`       | `grid-column-gap`       | `space`     |
| `gridColumn`          | `grid-column`           | none        |
| `gridRow`             | `grid-row`              | none        |
| `gridArea`            | `grid-area`             | none        |
| `gridAutoFlow`        | `grid-auto-flow`        | none        |
| `gridAutoRows`        | `grid-auto-rows`        | none        |
| `gridAutoColumns`     | `grid-auto-columns`     | none        |
| `gridTemplateRows`    | `grid-template-rows`    | none        |
| `gridTemplateColumns` | `grid-template-columns` | none        |
| `gridTemplateAreas`   | `grid-template-areas`   | none        |

### Layout

```js
function useCase() {
  return <Box width="full" height="2xl" overflow="hidden" />
}
```

#### Available Tokens

| Token           | CSS Property     | Theme Field |
| --------------- | ---------------- | ----------- |
| `w`             | `width`          | `sizes`     |
| `h`             | `height`         | `sizes`     |
| `minW`          | `min-width`      | `sizes`     |
| `maxW`          | `max-width`      | `sizes`     |
| `minH`          | `min-height`     | `sizes`     |
| `maxH`          | `max-height`     | `sizes`     |
| `display`       | `display`        | none        |
| `verticalAlign` | `vertical-align` | none        |
| `overflow`      | `overflow`       | none        |
| `overflowX`     | `overflowX`      | none        |
| `overflowY`     | `overflowY`      | none        |

#### Values

| Token Value    | Returned Value |
| -------------- | -------------- |
| 0              | `0rem`         |
| 1              | `0.0625rem`    |
| 2              | `0.125rem`     |
| 3              | `0.25rem`      |
| 4              | `0.5rem`       |
| 5              | `0.75rem`      |
| 6              | `1rem`         |
| 7              | `1.25rem`      |
| 8              | `1.5rem`       |
| 9              | `1.75rem`      |
| 10             | `2rem`         |
| 11             | `2.25rem`      |
| 12             | `2.5rem`       |
| 13             | `3rem`         |
| 14             | `4rem`         |
| 15             | `5rem`         |
| 16             | `6rem`         |
| 17             | `8rem`         |
| 18             | `10rem`        |
| 19             | `12rem`        |
| 20             | `14rem`        |
| 21             | `16rem`        |
| `xs`           | `20rem`        |
| `sm`           | `24rem`        |
| `md`           | `28rem`        |
| `lg`           | `32rem`        |
| `xl`           | `36rem`        |
| `2xl`          | `42rem`        |
| `3xl`          | `48rem`        |
| `4xl`          | `56rem`        |
| `5xl`          | `64rem`        |
| `6xl`          | `72rem`        |
| `1/2`          | `50%`          |
| `1/3`          | `33.333333%`   |
| `2/3`          | `66.666667%`   |
| `1/4`          | `25%`          |
| `2/4`          | `50%`          |
| `3/4`          | `75%`          |
| `1/5`          | `20%`          |
| `2/5`          | `40%`          |
| `3/5`          | `60%`          |
| `4/5`          | `80%`          |
| `1/6`          | `16.666667%`   |
| `2/6`          | `33.333333%`   |
| `3/6`          | `50%`          |
| `4/6`          | `66.666667%`   |
| `5/6`          | `83.333333%`   |
| `1/12`         | `8.333333%`    |
| `2/12`         | `16.666667%`   |
| `3/12`         | `25%`          |
| `4/12`         | `33.333333%`   |
| `5/12`         | `41.666667%`   |
| `6/12`         | `50%`          |
| `7/12`         | `58.333333%`   |
| `8/12`         | `66.666667%`   |
| `9/12`         | `75%`          |
| `10/12`        | `83.333333%`   |
| `11/12`        | `91.666667%`   |
| `full`         | `100%`         |
| `screenHeight` | `100vh`        |
| `screenWidth`  | `100vw`        |

### Border

```js
function useCase() {
  return <Box bw="1" br="3" bs="solid" />
}
```

#### Available Tokens

| Token | CSS Property          | Theme Field    |
| ----- | --------------------- | -------------- |
| `bw`  | `border-width`        | `borderWidths` |
| `btw` | `border-top-width`    | `borderWidths` |
| `bbw` | `border-bottom-width` | `borderWidths` |
| `brw` | `border-right-width`  | `borderWidths` |
| `blw` | `border-left-width`   | `borderWidths` |
| `br`  | `border-radius`       | `borderRadius` |
| `bs`  | `border-style`        | none           |
| `bts` | `border-top-style`    | none           |
| `bbs` | `border-bottom-style` | none           |
| `brs` | `border-right-style`  | none           |
| `bls` | `border-left-style`   | none           |

#### Values

| Token Value | Returned Value |
| ----------- | -------------- |
| 0           | `0rem`         |
| 1           | `0.0625rem`    |
| 2           | `0.125rem`     |
| 3           | `0.25rem`      |

### Typography

```js
function useCase() {
  return <Box fs="1" fv="regular" ta="center" lh="small" />
}
```

#### Available Tokens

| Token | CSS Property              | Theme Field     |
| ----- | ------------------------- | --------------- |
| `fs`  | `font-size`               | `fontSizes`     |
| `fv`  | `font-variation-settings` | `fontVariation` |
| `lh`  | `line-height`             | `lineHeights`   |
| `ta`  | `text-align`              | none            |

#### Values

##### Font size

| Token Value | Returned Value |
| ----------- | -------------- |
| 0           | `0.75rem`      |
| 1           | `0.875rem`     |
| 2           | `1rem`         |
| 3           | `1.125rem`     |
| 4           | `1.25rem`      |

##### Font Variation Settings

| Token Value | Returned Value |
| ----------- | -------------- |
| `light`     | `'wght' 80`    |
| `regular`   | `'wght' 92`    |
| `bold`      | `'wght' 100`   |

##### Line Height

| Token Value | Returned Value |
| ----------- | -------------- |
| `small`     | `1.125em`      |
| `body`      | `1.125em`      |
| `highlight` | `1.125em`      |
| `action`    | `1.5em`        |
| `subtitle`  | `1.5em`        |
| `headline`  | `1.5em`        |
