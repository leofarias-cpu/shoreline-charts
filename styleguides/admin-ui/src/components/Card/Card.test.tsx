import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import serializer from 'jest-emotion'

import { ThemeProvider } from '../../theme'
import { Card } from './index'

expect.addSnapshotSerializer(serializer)

describe('Button tests', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <Card>Example card</Card>
      </ThemeProvider>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
