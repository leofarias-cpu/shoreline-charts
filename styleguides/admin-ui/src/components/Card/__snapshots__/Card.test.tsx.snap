// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests should match snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 1.5rem;
  background-color: var(--theme-ui-colors-background,#FFFFFF);
  border: solid;
  border-color: var(--theme-ui-colors-muted-3,#D7DADF);
  border-width: 1px;
  border-radius: 4px;
}

<div
    class="emotion-0"
  >
    Example card
  </div>
</DocumentFragment>
`;
